project/agent-groups/components/AgentGroupsList.vue

```diff
@@ -0,0 +1,281 @@
+  loading: boolean
+  agentGroupService: AgentGroupService
+  agentGroups: any[]
+  headers: { title: string, align: string, sortable: boolean, key: string, width: string }[]
+  agentGroup: any | null
+  itemsPerPage: number
+  editorLanguage: string
+  editorOptions: object
+  deleteAgentGroupDialog: boolean
+  onActionAgentGroup: any
+  showForm: boolean
+  selectedAgentGroup: any | null
+}
+interface UpdatedAgentGroup {
+  description?: string
+  id: string
+  key?: string
+  name?: string
+  yamlSource?: string
+  scanner: Scanner
+}
+
+export default defineComponent({
+  components: {
+    DfConfirmationModal,
+    AgentGroupForm
+  },
+  data(): Data {
+    return {
+      loading: true,
+      agentGroupService: new AgentGroupService(this.$axios),
+      agentGroups: [],
+      headers: [
+        { title: 'Key', align: 'left', sortable: false, key: 'key', width: '24%' },
+        { title: 'Name', align: 'left', sortable: false, key: 'name', width: '14%' },
+        { title: 'Description', align: 'left', sortable: false, key: 'description', width: '14%' },
+        { title: 'Actions', align: 'left', sortable: false, key: 'actions', width: '5%' }
+      ],
+      agentGroup: null,
+      onActionAgentGroup: null,
+      itemsPerPage: 15,
+      editorLanguage: 'yaml',
+      editorOptions: {
+        theme: 'vs',
+        wordWrap: 'on',
+        wordWrapColumn: 'on',
+        fontFamily: 'Fira Code',
+        automaticLayout: true,
+        minimap: { enabled: false }
+      },
+      deleteAgentGroupDialog: false,
+      showForm: false,
+      selectedAgentGroup: null
+
+    }
+  },
+  computed: {
+    ...mapState(useScannersStore, ['scanners']),
+    actions(): Array<ActionsType> {
+      return [
+        {
+          title: 'Edit',
+          action: (agentGroup) => this.updateAgentGroup(agentGroup),
+          icon: 'mdi-pencil-outline',
+          color: 'primary',
+          divider: true
+        },
+        {
+          title: 'Delete',
+          action: (agentGroup) => this.deleteAgentGroup(agentGroup),
+          icon: 'mdi-delete',
+          color: 'red'
+        }
+      ]
+    }
+  },
+  mounted() {
+    this.fetchAgentGroups()
+  },
+  methods: {
+    ...mapActions(useNotificationsStore, ['reportSuccess', 'reportError']),
+
+    /**
+     * Fetches the list of agent groups from the server and updates the component state.
+     */
+    fetchAgentGroups(): void {
+      this.agentGroups = []
+      const promises = this.scanners.map((scanner) =>
+        this.agentGroupService.getAgentGroups(scanner).then((val) => {
+          if (Array.isArray(val)) {
+            this.agentGroups.push(...val.map((agentGroup) => ({ ...agentGroup, scanner })))
+          }
+        })
+      )
+
+      Promise.all(promises)
+        .then(() => {
+          this.loading = false
+        })
+        .catch((error) => {
+          console.error('Error fetching agent groups:', error)
+          this.loading = false
+        })
+    },
```

Comment:
Use `Promise.allSettled`. With `Promise.all` if one request fails, the others won't execute, and that's not what we want. + using `async/await` is cleaner than `.then /.catch`